stock_quantity_history_location/static/src/components/inventory_report/inventory_report.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates>
+    <t t-inherit="stock.InventoryReport.Buttons" t-inherit-mode="extension" owl="1">
+        <xpath expr="//button[contains(@t-if, 'stock.quant')]" position="attributes">
+            <attribute name="t-if" add="and !this.multi_location" separator=" " />
```

Comment:
There's no safe way to add conditions to an attribute, as order is not guaranteed

I'd suggest creating a `<t t-if` element and moving button there, and inserting the new button just after that element using `t-else`